lib/elixir/src/elixir_erl_pass.erl

```diff
@@ -211,31 +211,37 @@ translate({Name, Meta, Args}, S) when is_atom(Name), is_list(Meta), is_list(Args
 
 %% Remote calls
 
-translate({{'.', _, [Left, Right]}, Meta, []}, S)
+translate({{'.', _, [Left, Right]}, Meta, []}, #elixir_erl{context=Context} = S)
     when is_tuple(Left), is_atom(Right), is_list(Meta) ->
   {TLeft, SL}  = translate(Left, S),
-  {Var, SV} = elixir_erl_var:build('_', SL),
-
   Ann = ?ann(Meta),
-  Generated = erl_anno:set_generated(true, Ann),
   TRight = {atom, Ann, Right},
-  TVar = {var, Ann, Var},
-  TError = {tuple, Ann, [{atom, Ann, badkey}, TRight, TVar]},
-
-  {{'case', Generated, TLeft, [
-    {clause, Generated,
-      [{map, Ann, [{map_field_exact, Ann, TRight, TVar}]}],
-      [],
-      [TVar]},
-    {clause, Generated,
-      [TVar],
-      [[?remote(Generated, erlang, is_map, [TVar])]],
-      [?remote(Ann, erlang, error, [TError])]},
-    {clause, Generated,
-      [TVar],
-      [],
-      [{call, Generated, {remote, Generated, TVar, TRight}, []}]}
-  ]}, SV};
+
+  case lists:keyfind(no_parens, 1, Meta) of
+    {no_parens, true} when Context == guard andalso is_tuple(Left) ->
```

Comment:
Given we validate it during expansion (great job there btw), then we don't need to check for the tuple format nor the `Meta`. :) You can just see if context is a `guard` and then emit the guard code!